Tasks/SonarQubePostTest/SonarQubeMetrics.ps1

```diff
@@ -38,18 +38,38 @@ function WaitForAnalysisToFinish
     Assert (![String]::IsNullOrEmpty($analysisId)) "Could not fetch the analysis id"
 }
 
+
+function FetchQualityGateDetails
+{
+    param ([Array]$errors,[Array]$warnings)
+    
+    $analysisId = GetTaskContextVariable "MSBuild.SonarQube.AnalysisId"             
+    Assert (![String]::IsNullOrEmpty($analysisId)) "WaitForAnalysisToFinish should be called first."
+       
+    $response = InvokeGetRestMethod "/api/qualitygates/project_status?analysisId=$analysisId" $true
```

Comment:
Calling WaitForAnalysisToComplete on an early SQ will throw a friendly error message. Currently this is only possible for the build breaker. In case of the summary report the old "legacy" report is uploaded if SQ version is 5.2 or lower so WaitForAnalysisToComplete won't even be called. 

But I understand your underlying question - as I call more and more APIs with different version requirements there should be a generic mechanism to handle the version check. I'll refactor fetching the metrics next time I need to call a metric requiring a version check.